std/datetime/systime.d

```diff
@@ -8560,6 +8630,7 @@ public:
     // bug# 17801
     @safe unittest
     {
+        import core.time;
```

Comment:
How about a top-level `version (unittest) import core.time;` instead? Maybe even just a selective one?
It's indeed annoying that privately imported modules are resolved eagerly, albeit none of their symbols might ever be needed.

Reply:
I considered it but am weary of the bugs that could happen. Consider:

```d
version(unittest) import std.stdio;
void fun(T)(T value)
{
    writeln(value);
}
unittest
{
    fun(5);
}
```

This module compiles on its own, passes unittests, is importable when not used, but issues an error whenever the user attempts to use `fun`. I think we should make it a guideline that `version(unittest)` top-level imports should be used with caution.

Reply:
FYI: At least if it's a publicly documented test, this can't happen as we execute them separately exactly for the reason to protect against these kinds of bugs.

Reply:
> I think we should make it a guideline that version(unittest) top-level imports should be used with caution.

It's more or less been a guideline for a while now, precisely because it caused problems in the past. I don't know that we've actually written it down anywhere though, just told people not to do it when it comes up in PRs.